experimental/detect_breakage.py

```diff
@@ -0,0 +1,80 @@
+import enum
+import importlib
+import inspect
+import pkgutil
+
+def get_package_info(pkgname):
+    """returns a dict containing info about modules, classes, functions"""
+    pkg = importlib.import_module(pkgname)
+    iter_modules = pkgutil.iter_modules(pkg.__path__)
+    modules = []
+
+    for (_, name, _) in iter_modules:
+        module = importlib.import_module('%s.%s' % (pkgname, name))
+        modules.append(module)
+
+    res = get_module_info(modules)
+    return res
+
+def get_module_info(modules):
+    """returns a dict containing info on classes and functions"""
+    res = {}
+    for module in modules:
+        if inspect.ismodule(module):
+            modname = module.__name__
+            res[modname] = {}
+
+            classes = inspect.getmembers(module, inspect.isclass)
+            classes = [c for c in classes if c[1].__module__ == modname]
+            functions = inspect.getmembers(module, inspect.isfunction)
+            functions = [f for f in functions if f[1].__module__ == modname]
+
+            res[modname]['classes'] = get_class_info(classes)
+            res[modname]['functions'] = get_function_info(functions)
+    return res
+
+def get_class_info(classes):
+    """returns a dict containing info on args, subclasses and functions"""
+    res = {}
+    for name, value in classes:
+        if name[0] != '_':
+            res[name] = {}
+            # TODO: expand this - enum probably isn't the only exception
+            if issubclass(value, enum.Enum):
+                args = []
+            else:
+                try:
+                    args = inspect.getfullargspec(value).args
+                except TypeError:
```

Comment:
Could you explain in a comment why this is necessary?

Reply:
Since inconsistency in the class args among different non major versions would be considered a breaking change, I'm using `getfullargspec` function get get those args for a given class. However, there are some classes that don't take in any args ([example](https://github.com/ylil93/cloud-opensource-python/blob/81233b1df3d1aaf1001e3ddc9f7f324fd127e9fe/compatibility_lib/compatibility_lib/dependency_highlighter.py#L28)) and this function will throw a TypeError in those cases, so I wrote an `except` as a workaround and a way of indicating that no args are expected.